sstables/sstable_set.cc

```diff
@@ -178,6 +178,21 @@ sstable_set::make_incremental_selector() const {
     return incremental_selector(_impl->make_incremental_selector(), *_schema);
 }
 
+template <typename F>
+class func_selector : public incremental_selector_impl {
+    F _f;
+public:
+    func_selector(F f) : _f(std::move(f)) {}
+    virtual std::tuple<dht::partition_range, std::vector<shared_sstable>, dht::ring_position_view> select(const dht::ring_position_view& rpv) override {
+        return _f(rpv);
+    }
+};
+
+template <typename F>
+std::unique_ptr<func_selector<F>> make_func_selector(F&& f) {
+    return std::make_unique<func_selector<F>>(std::forward<F>(f));
+}
```

Comment:
What's wrong with just providing implementations of virtual methods? Why introduce yet another dynamic dispatch, just to be able to do it with a lambda?

Reply:
no more `make_func_selector`